lib/vtls/openssl.c

```diff
@@ -2029,6 +2029,20 @@ set_ssl_version_min_max(long *ctx_options, struct connectdata *conn,
   return CURLE_OK;
 }
 
+static int _verify_no_date_check(int preverify_ok, X509_STORE_CTX *x509_ctx)
+{
+  /* If we have an error related to the validity period, then unset the error.
+     and return success */
+  if(X509_STORE_CTX_get_error(x509_ctx) == X509_V_ERR_CERT_HAS_EXPIRED ||
```

Comment:
Does it make sense to check this even if `preverify_ok` is already false? Maybe prepend `preverify_ok &&` to the condition?

Reply:
preverify_ok will be 0 in the case that we care about here. That is, preverify_ok will be 0 if X509_STORE_CTX_get_error(x509_ctx) == X509_V_ERR_CERT_HAS_EXPIRED or X509_STORE_CTX_get_error(x509_ctx) == X509_V_ERR_CERT_NOT_YET_VALID.

Reply:
I think that you are correct. We should probably add something like `!preverify_ok && ...` to save the call to  `X509_STORE_CTX_get_error()` if `preverify_ok==1`. I'll work on that tomorrow.